modules/channel-twilio/src/backend/client.ts

```diff
@@ -0,0 +1,265 @@
+    if (index && (await this.handleIndexReponse(index - 1, from, to))) {
+      return
+    }
+
+    await this.kvs.delete(from)
+
+    await this.bp.events.sendEvent(
+      this.bp.IO.Event({
+        botId: this.botId,
+        channel: 'twilio',
+        direction: 'incoming',
+        type: 'text',
+        payload: {
+          type: 'text',
+          text: text
+        },
+        threadId: to,
+        target: from
+      })
+    )
+  }
+
+  async handleIndexReponse(index: number, from: string, to: string): Promise<boolean> {
+    if (!(await this.kvs.exists(from))) {
+      return
+    }
+
+    const options = await this.kvs.get(from)
+    const option = options[index]
+    if (!option) {
+      return
+    }
+
+    if (option.type === 'url') {
+      return true
+    }
+
+    await this.kvs.delete(from)
+
+    await this.bp.events.sendEvent(
+      this.bp.IO.Event({
+        botId: this.botId,
+        channel: 'twilio',
+        direction: 'incoming',
+        type: option.type,
+        payload: {
+          type: option.type,
+          text: option.type === 'say_something' ? option.value : option.label,
+          payload: option.value
+        },
+        threadId: to,
+        target: from
+      })
+    )
+
+    return true
+  }
+
+  async handleOutgoingEvent(event: sdk.IO.Event, next: sdk.IO.MiddlewareNextCallback) {
+    if (event.type === 'text') {
+      await this.sendText(event)
+    } else if (event.type === 'file') {
+      await this.sendImage(event)
+    } else if (event.type === 'carousel') {
+      await this.sendCarousel(event)
+    } else if (event.payload.quick_replies) {
+      await this.sendChoices(event)
+    } else if (event.payload.options) {
+      await this.sendDropdown(event)
+    }
+
+    next(undefined, false)
+  }
+
+  async sendText(event: sdk.IO.Event) {
+    await this.sendMessage(event, {
+      body: event.payload.text
+    })
+  }
+
+  async sendImage(event: sdk.IO.Event) {
+    await this.sendMessage(event, {
+      body: event.payload.title,
+      mediaUrl: [event.payload.url]
+    })
+  }
+
+  async sendCarousel(event: sdk.IO.Event) {
+    for (const { subtitle, title, picture, buttons } of event.payload.elements) {
+      let body = `${title}\n\n`
+      if (subtitle) {
+        body += `${subtitle}`
+      }
+
+      const options: MessageOption[] = []
+      for (const button of buttons) {
+        if (button.type === 'open_url') {
+          options.push({
+            label: `${button.title} : ${button.url}`,
+            value: undefined,
+            type: 'url'
+          })
+        } else if (button.type === 'postback') {
+          options.push({
+            label: button.title,
+            value: button.payload,
+            type: 'postback'
+          })
+        } else if (button.type === 'say_something') {
+          options.push({
+            label: button.title,
+            value: button.text,
+            type: 'say_something'
+          })
```

Comment:
With deconstructor & putting your object on a single line, this block of code will be much smaller